README.md

```diff
@@ -1 +1,1380 @@
+
+This writing aims at a quick introduction to scripting across the Platform—via comparing the scripting environments, references to the official documentation, and an example of scripting in each of the three products.
+
+## Contents
+
+* [Conclusion](#conclusion)
+* [Summary](#summary)
+* [A Simple Example of Scripting in ForgeRock Components](#example)
+    * [AM](#example-am)
+    * [IDM](#example-idm)
+    * [IG](#example-ig)
+* [References](#references)
+
+## <a id="conclusion"></a>Conclusion
+
+[Back to the Top](#top)
+
+The scripting objectives and implementation are driven by the product's functionality and the environment it provides. Hence, the scripts' location, configuration, security, the data and methods a script can use, and the way the scripts are managed are specific to a product.
+
+There are certain similarities as well: the choice of scripting languages, ability to access the underlying Java functionality and the context data, logging methods, access to the request object, and ability to make back-channel HTTP requests—all converge into a similar experience at certain level. The script configuration and content for all three products could be represented as JSON and saved in the file system.
+
+The ability of scripts to communicate with external network resources is a powerful tool. The current security measures do not seem to apply restrictions to endpoints a script may access, which may represent an additional concern about the script's overall security if the administrative UI/API protections can be breached. However, because the same administrative privileges that allow manage scripts will likely allow to change other configuration, the network restriction policies may need to be implemented outside of the product configuration.
+
+Scripts add flexibility to the ForgeRock Identity Platform. While a script might not be performing at the same level as a native/standard implementation, custom scripts can be used to substitute functionality that is not yet present in the software or is specific to a certain deployment.
+
+## <a id="summary"></a>Summary
+
+[Back to the Top](#top)
+
+This section is a briefed overview of different scripting aspects in the three products.
+
+The [References](#references) section contains collection of links to the official scripting documentation. The links are organized by product and by area of concern.
+
+* ### <a id="summary-application-and-environment"></a>Application and Environment
+
+    Scripting application is augmenting or extending native functionality of a product. The extension points could be events associated with an object or a procedure; for example, an update event on a managed object in IDM. Or, the scripts could represent an essential part of a procedure, necessary for its successful completion; for example, scripts employed in AM authentication flows.
```

Comment:
I think in pretty much every case that a script is used, its use is considered "an essential part of a procedure, necessary for its successful completion"

Reply:
How about this:
> The extension points could be events associated with an object or a procedure; for example, an update event on a managed object in IDM. Or, the scripts could represent a procedure; for example, scripts performing authentication in AM.

I am trying to draw distinction between an action and an event associated with action.